### PR TITLE
dx: add CONTRIBUTING.md file

### LINKED ISSUE #71: [DX] <Add a Contriubting.md file explaining all the contributing guidelines>
### Description

Try to add a contriubting.md file explaining all the contributing guidelines.

### PR DESCRIPTION
**Pull request by jideabdqudus**
Closes #71 

- Created and wrote the Contributing Doc to assist with developers experience.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Contributing
+
+Defaang lives through contributions from the community, In contributing to this repository, please first discuss the change you wish to make via issue, discord, or any other method with the owners of this repository before making a change.
```
Comment (ykdojo) [inline CONTRIBUTING.md:3]: Defaang -> defaang

first commma -> period

via issue -> via an issue

owners -> maintainers

**Review by ykdojo — CHANGES_REQUESTED** [review]
This is a good start, but it's a lot to review in one go.

Any chance you could break it down to much smaller pieces and send individual PRs for them?

**Comment by jideabdqudus** [comment]
> This is a good start, but it's a lot to review in one go.
> 
> Any chance you could break it down to much smaller pieces and send individual PRs for them?

Do I close this PR and send new PRs as individual chunks? Or how do I go about this

**Comment by rohitdasu** [comment]
> > This is a good start, but it's a lot to review in one go.
> > Any chance you could break it down to much smaller pieces and send individual PRs for them?
> 
> Do I close this PR and send new PRs as individual chunks? Or how do I go about this

Hi @jideabdqudus you don't need to close this PR. Update the PR whichever changes requested by @ykdojo

**Comment by subhoghoshX** [comment]
LGTM. Make the changes that YK mentioned and make sure to run it through a grammar checker like Grammarly or Quillbot.

**Comment by jideabdqudus** [comment]
> LGTM. Make the changes that YK mentioned and make sure to run it through a grammar checker like Grammarly or Quillbot.

Changes have been made & I did the spell check as well cc @rohitdasu @ykdojo

File: CONTRIBUTING.md (hunk 1 above)
Comment (jideabdqudus) [reply CONTRIBUTING.md]: Done!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+An active community forms the backbone of every major Open Source project. The community is present on various platforms on the internet, where we actively help each other out – not only with regard to using the application. GitHub is the most tech-focused part of the community, albeit [discussing here](https://github.com/ykdojo/defaang/discussions) is the way to go for more theoretical discussions. Furthermore, we have a [Discord-server](https://discord.gg/aJp6Fypb) where we discuss building the app and help each other out.
+
+### All Communication Channels
+
+- [Discord](https://discord.gg/aJp6Fypb)
+- [GitHub](https://github.com/ykdojo/defaang/discussions)
+
+## How to Help
+
+If you have some experiences with developing, we would gladly welcome your help in fixing issues and adding features. Even if you just start: There are tons of smaller things and chores to do, so by cleaning up code here and there you can improve your skills and also help the project. Even if it's a minor fix, it is still a fix!
```
Comment (subhoghoshX) [inline CONTRIBUTING.md:28]: experiences -> experience

Comment (jideabdqudus) [reply CONTRIBUTING.md]: Fixed :)

**Review by subhoghoshX — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  - [How to Help](#how-to-help)
+    - [Our Stack (with linked documentation)](#our-stack)
+    - [Issue Labels](#issue-labels)
+    - [Semantic Commits](#semantic-commits)
+    - [Commit Example](#commit-example)
+  - [Remarks](#remarks)
+
+## Where to Get Help
+
+An active community forms the backbone of every major Open Source project. The community is present on various platforms on the internet, where we actively help each other out – not only with regard to using the application. GitHub is the most tech-focused part of the community, albeit [discussing here](https://github.com/ykdojo/defaang/discussions) is the way to go for more theoretical discussions. Furthermore, we have a [Discord-server](https://discord.gg/aJp6Fypb) where we discuss building the app and help each other out.
```
Comment (ykdojo) [inline CONTRIBUTING.md:19]: I think this paragraph is too complex. Can you simplify?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+### All Communication Channels
+
+- [Discord](https://discord.gg/aJp6Fypb)
+- [GitHub](https://github.com/ykdojo/defaang/discussions)
+
+## How to Help
+
+If you have some experience with developing, we would gladly welcome your help in fixing issues and adding features. Even if you just start: There are tons of smaller things and chores to do, so by cleaning up code here and there you can improve your skills and also help the project. Even if it's a minor fix, it is still a fix!
```
Comment (ykdojo) [inline CONTRIBUTING.md:28]: I really don't like the word chores here - I really don't think we have any *chores* to do? I think every task is interesting in a different way

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+### Our Stack
+
+- Frontend: [NextJS / React](https://nextjs.org/)
+- CSS framework: [Tailwind](https://tailwindcss.com/)
+- Backend: [Supabase](https://supabase.com/)
+- Hosting: [Vercel](https://vercel.com/)
+
+### Issue Labels
+
+There are – broadly speaking – three categories of issues on the defaang GitHub issue tracker, which you can distinguish with their associated labels.
```
Comment (ykdojo) [inline CONTRIBUTING.md:45]: simplify this sentence

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+3. **Features** are bigger additions to the application that require a completely new feature.
+
+If you want to help, make sure to _communicate your intent_, that is:
+
+1. If you want to tackle an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
+2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
+3. If you have a PR and are not completely sure about whether it's right, or not, just open the PR and outline your questions there. It is much easier to work with something we can see and improve the PR than if you ask without something concrete.
+
+In any case, just remember to just open PRs and dive in. The worst that could happen is that we ask you to rewrite some file, but that has never happened and is pretty unlikely to happen in the future.
```
Comment (ykdojo) [inline CONTRIBUTING.md:57]: say "just" once here

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+style: (formatting, missing semi colons, etc; no production code change)
+refactor: (refactoring production code, eg. renaming a variable)
+test: (adding missing tests, refactoring tests; no production code change)
+dx: (dev experience; anything that helps to improve developers' experience)
+chore: (updating grunt tasks etc; no production code change)
+```
+
+#### Commit Example
+
+```bash
```
Comment (ykdojo) [inline CONTRIBUTING.md:76]: I don't think this should be rendered as a bash command

**Review by ykdojo — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 7 above)
Comment (jideabdqudus) [reply CONTRIBUTING.md]: I've made the new changes across the PR

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
+# Contributing
+
+defaang lives through contributions from the community. In contributing to this repository, please first discuss the change you wish to make via an issue, discord, or any other method with the maintainers of this repository before making a change.
```
Comment (ykdojo) [inline CONTRIBUTING.md:3]: In contributing -> When contributing

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+**Table of Contents**
+
+- [Contributing](#contributing)
```
Comment (ykdojo) [inline CONTRIBUTING.md:7]: do these links work?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+2. **Enhancements** are some small additions that require adding some more functionality to a feature, but that should not evolve into an own distinct feature. Even if the issue description reads like a full feature to you, the "enhancement" label means there is already said functionality, which just needs to be amended. Please ask if you are unsure.
+3. **Features** are bigger additions to the application that require a completely new feature.
+
+If you want to help, make sure to _communicate your intent_, that is:
+
+1. If you want to tackle an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
+2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
```
Comment (ykdojo) [inline CONTRIBUTING.md:54]: If you want to tackle an issue -> If you start working on an issue

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+3. If you have a PR and are not completely sure about whether it's right, or not, just open the PR and outline your questions there. It is much easier to work with something we can see and improve the PR than if you ask without something concrete.
+
+In any case, just remember to just open PRs and dive in. The worst that could happen is that we ask you to rewrite some file, but that has never happened and is pretty unlikely to happen in the future.
```
Comment (ykdojo) [inline CONTRIBUTING.md:57]: I'd remove this part:

> The worst that could happen is that we ask you to rewrite some file, but that has never happened and is pretty unlikely to happen in the future.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+#### Commit Example
+
+feat: added authentication using supabase auth
+
+[Learn more about semantic commits](https://www.conventionalcommits.org/en/v1.0.0/)
+
+## Remarks
+
+If something is missing here, or you feel something is not well described, either directly open a PR or ask one of the community places.
```
Comment (ykdojo) [inline CONTRIBUTING.md:82]: ask *in* one of the...

**Review by ykdojo — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 9 above)
Comment (jideabdqudus) [reply CONTRIBUTING.md:7]: Yeah

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -50,11 +50,11 @@ There are three categories of issues on the defaang GitHub issue tracker:
 
 If you want to help, make sure to _communicate your intent_, that is:
 
-1. If you want to tackle an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
-2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
+1. If you want to start working on an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
```
Comment (ykdojo) [inline CONTRIBUTING.md:53]: I think *tackle* here is ok? I just thought too much repeating won't sound good

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -50,11 +50,11 @@ There are three categories of issues on the defaang GitHub issue tracker:
...
+2. If you start working on an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
```
Comment (ykdojo) [inline CONTRIBUTING.md:54]: plus, here, we're talking about the point at which people have already started - so that's the difference

**Comment by ykdojo** [comment]
^Once you address the comments I added above, I think it looks good to me.

I'll wait for one or two more reviewers to check it, though.

**Comment by iShibi** [comment]
You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.

Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.

File: CONTRIBUTING.md (hunk 13 above)
Comment (jideabdqudus) [reply CONTRIBUTING.md:53]: Okay, that makes sense.

**Comment by jideabdqudus** [comment]
> You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> 
> Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.

Can you give more context to what you mean here?

**Comment by veganaize** [comment]
> > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> > Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.
> 
> Can you give more context to what you mean here?

@jideabdqudus I think they mean you should move the _contributing_ related stuff, from `README.md`, into `CONTRIBUTING.md`.

For their second request, about creating a new feature branch for each PR, you might reference the following document:
[https://github.com/LibreSprite/LibreSprite/blob/master/CONTRIBUTING.md#making-changes]()

**Comment by jideabdqudus** [comment]
> > > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> > > Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.
> > 
> > 
> > Can you give more context to what you mean here?
> 
> @jideabdqudus I think they mean you should move the _contributing_ related stuff, from `README.md`, into `CONTRIBUTING.md`.
> 
> For their second request, about creating a new feature branch for each PR, you might reference the following document: https://github.com/LibreSprite/LibreSprite/blob/master/CONTRIBUTING.md#making-changes

I got this,

For the first request, stuff from README.md are already in here. Then I added content explaining the steps in making changes.

**Comment by veganaize** [comment]
> > > > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.

> For the first request, stuff from README.md are already in here. Then I added content explaining the steps in making changes.

@jideabdqudus Can you remove the entire "How to contribute" section from `README.md` and place it inside of `CONTRIBUTING.md` and push that?

**Comment by jideabdqudus** [comment]
> > > > > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> 
> > For the first request, stuff from README.md are already in here. Then I added content explaining the steps in making changes.
> 
> @jideabdqudus Can you remove the entire "How to contribute" section from `README.md` ~and place it inside of `CONTRIBUTING.md`~, link to `CONTRIBUTING.md` from `README.md`, and push that?

Okay that makes sense, I just did with the new push

**Comment by veganaize** [comment]
> > @jideabdqudus Can you remove the entire "How to contribute" section from `README.md` ~and place it inside of `CONTRIBUTING.md`~, link to `CONTRIBUTING.md` from `README.md`, and push that?
> 
> Okay that makes sense, I just did with the new push

I could be wrong but it looks like you've removed the "How to contribute" section from `CONTRIBUTING.md` and not from `README.md`?  It should be the other way around.

**Comment by jideabdqudus** [comment]
> > > @jideabdqudus Can you remove the entire "How to contribute" section from `README.md` ~and place it inside of `CONTRIBUTING.md`~, link to `CONTRIBUTING.md` from `README.md`, and push that?
> > 
> > 
> > Okay that makes sense, I just did with the new push
> 
> I could be wrong but it looks like you've removed the "How to contribute" section from `CONTRIBUTING.md` and not from `README.md`? It should be the other way around. [_Sorry if my crossed out comment confused you._]

Oh Dang, you lost me a bit. I've reverted this and fixed README.md

**Comment by veganaize** [comment]
> Oh Dang, you lost me a bit. I've reverted this and fixed README.md

_Sorry... my bad @jideabdqudus !_

This all looks good, to me.  What do you think @ykdojo , @rohitdasu , and @iShibi ?

**Review by ykdojo — CHANGES_REQUESTED** [review]
Can you add a link to CONTRIBUTING.md from README.md?

**Comment by rohitdasu** [comment]
> Can you add a link to CONTRIBUTING.md from README.md?

yes agree, should have a link to CONTRIBUTING.md

**Comment by jideabdqudus** [comment]
Okay guys, back to this :).
I've added a link to CONTRIBUTING.md from README.md with the latest push. 
Can we once go through the file so I wouldn't have to make little changes again @ykdojo @rohitdasu @iShibi

**Review by ykdojo — APPROVED** [review]

**Comment by ykdojo** [comment]
Thank you so much! Looks good to me.

**Comment by iShibi** [comment]
@jideabdqudus looks good. Resolve the git conflicts by rebasing and then we will merge it.

**Comment by jideabdqudus** [comment]
> @jideabdqudus looks good. Resolve the git conflicts by rebasing and then we will merge it.

Feel free to merge now

**Comment by iShibi** [comment]
You overwrote the changes made in the `main`. I'll have to rebase it and then we can merge.

**Review by iShibi — APPROVED** [review]

**Pull request merged**